src/FirebaseProject.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kreait\Laravel\Firebase;
+
+use Kreait\Firebase;
+
+class FirebaseProject
+{
+    /** @var \Kreait\Firebase\Factory $factory */
+    protected $factory;
+
+    /** @var array $config */
+    protected $config;
+
+    /** @var \Kreait\Firebase\Auth|null $auth */
+    protected $auth = null;
+
+    /** @var \Kreait\Firebase\Database|null $database */
+    protected $database = null;
+
+    /** @var \Kreait\Firebase\DynamicLinks|null $dynamicLinks */
+    protected $dynamicLinks = null;
+
+    /** @var \Kreait\Firebase\Firestore|null $firestore */
+    protected $firestore = null;
+
+    /** @var \Kreait\Firebase\Messaging|null $messaging */
+    protected $messaging = null;
+
+    /** @var \Kreait\Firebase\RemoteConfig|null $remoteConfig */
+    protected $remoteConfig = null;
+
+    /** @var \Kreait\Firebase\Storage|null $storage */
+    protected $storage = null;
+
+    public function __construct(Firebase\Factory $factory, array $config)
+    {
+        $this->factory = $factory;
+        $this->config = $config;
+    }
+
+    public function auth(): Firebase\Auth
+    {
+        if (! $this->auth) {
```

Comment:
```suggestion
        if (!$this->auth) {
```
Again, this is just style, but this is something I would otherwise change in a future commit, or as soon as I introduce PHP CS Fixer (with similar rules as in `kreait/firebase-php`)